### PR TITLE
feat: clarity syntax highlighting

### PR DESCRIPTION
**Pull request by hugocaillard**
## Description

Add Clarity syntax highlighting.
Porting the changes from https://github.com/stacks-network/docs/pull/1576

Screenshot 2023-02-24 at 17 11 32
(View production version of [this page](https://docs.hiro.so/example-apps/billboard#review-billboard-contract))

## Checklist

- [x] [Conventional commits were used](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,8 +32,8 @@
     "docusaurus-plugin-segment": "^1.0.3",
     "file-loader": "^6.2.0",
     "prism-react-renderer": "^1.2.1",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0",
+    "react": "^17.0.2",
```
Comment (hugocaillard) [inline package.json:35]: Docusaurus doesn't support react@18 for now

**Comment by hugocaillard** [comment]
Requesting a review from you @sabbyanandan since you seem to be reviewing most PRs on this repo. Please assign someone else if it makes sense.
I have broken links issues on the lintcheck but definitely not from this PR. I fixed (url changed on the stacks.co docs website), for the other, the page really doesn't exist and I don't know if it has been moved somewhere else 🤔 (`https://docs.stacks.co/docs/write-smart-contracts/values`)

File: package.json (hunk 1 above)
Comment (sabbyanandan) [reply package.json:35]: Interesting; [this commit](https://github.com/hirosystems/docs/commit/59bdb65cee4fa40a32ec18c3781f34817cf92e5b) introduced the dependency upgrades. Maybe @janniks would know more to answer this.

File: docs/stacks.js/how-to-guides/how-to-migrate-stacking-to-2.1.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ title: Migrate Stacking to 2.1
 
 #### "Stacks 2.1"
 
-[Stacks 2.1](https://stacks.org/stacks-21-what-to-expect) is a fork and protocol upgrade of the Stacks blockchain. Among other improvements, it includes an updated [Proof-of-Transfer](https://docs.stacks.co/docs/understand-stacks/proof-of-transfer) (PoX) [contract](https://docs.stacks.co/docs/noteworthy-contracts/stacking-contract) (aka PoX-2) for Stacking.
+[Stacks 2.1](https://stacks.org/stacks-21-what-to-expect) is a fork and protocol upgrade of the Stacks blockchain. Among other improvements, it includes an updated [Proof-of-Transfer](https://docs.stacks.co/docs/understand-stacks/proof-of-transfer) (PoX) [contract](https://docs.stacks.co/docs/clarity/noteworthy-contracts/stacking-contract) (aka PoX-2) for Stacking.
```
Comment (sabbyanandan) [inline docs/stacks.js/how-to-guides/how-to-migrate-stacking-to-2.1.md:11]: Good catch!

**Review by sabbyanandan — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (hugocaillard) [reply package.json:35]: I don't think it break anything, just displays a warning

**Comment by hugocaillard** [comment]
@sabbyanandan I just merged this PR since you approved. 
But I just saw that you assigned @trichards-hiro. I hope it was ok that I merged it

**Pull request merged**